pym/bob/cmds/build.py

```diff
@@ -980,20 +1008,62 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                     self._runShell(packageStep, "package")
                     packageHash = hashWorkspace(packageStep)
                     audit = self._generateAudit(packageStep, depth, packageHash)
+                    # copy to shared location if possible
+                    self.__installSharedPackage(packageBuildId, packageStep, prettyPackagePath)
                     workspaceChanged = True
                     self.__statistic.packagesBuilt += 1
                     if packageBuildId and self.__archive.canUploadLocal():
-                        self.__archive.uploadPackage(packageBuildId, audit, prettyPackagePath, self.__verbose)
+                        if packageStep.isRelocatable():
```

Comment:
That should be replaced by `if canUpDownload and self.__archive.canUploadLocal():` instead of the nested `if`.